### ISSUE TITLE
RigidBodyPropellor does not work if the floating base is using quaternion

### ISSUE THREAD

**Issue by hongkai-dai** [RobotLocomotion/drake#1132]
In `RigidBodyPropeller`, when computing the force in the generalized coordinate due to the thrus, it calls
```
B_mod(:,obj.input_num) = obj.scale_factor_thrust*J'*axis_world;
```
and `J` is computed from forwardKin. But when using quaternion and twist, we should call geometricJacobian to compute J.

**Comment by hongkai-dai** [comment]
@RussTedrake , can you assign this to me?

**Comment by RussTedrake** [comment]
resolved in #1375
